src/init/features/apphosting/index.ts

```diff
@@ -62,7 +66,30 @@ export async function doSetup(
     }
   }
 
-  logBullet("First we need a few details to create your backend.");
+  logBullet("First we need a few details to create your backend.\n");
+
+  let webAppId: string;
+
+  const existingUserProjectWebApps = new Map(
+    (await listFirebaseApps(projectId, AppPlatform.WEB)).map((obj) => [
+      // displayName can be null, use name instead if so. Example - displayName: "mathusan-web-app", name: "projects/mathusan-fwp/webApps/1:461896338144:web:426291191cccce65fede85"
+      obj.displayName ?? obj.name,
+      obj.appId,
+    ]),
+  );
+
+  if (webAppName) {
+    if (!existingUserProjectWebApps.get(webAppName)) {
+      throw new FirebaseError(`The web app '${webAppName}' does not exist in project ${projectId}`);
```

Comment:
Let's move all of this to a helper so that the top level control flow remains clear